webofneeds/won-core/src/main/java/won/cryptography/service/keystore/KeyStoreService.java

```diff
@@ -0,0 +1,43 @@
+package won.cryptography.service.keystore;
+
+import java.io.IOException;
+import java.security.KeyPair;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.cert.Certificate;
+
+public interface KeyStoreService {
+
+	PrivateKey getPrivateKey(String alias);
+
+	/**
+	 * Returns the public key of the certificate stored under the specified alias or
+	 * null if no such certificat is found.
+	 *
+	 * @param alias
+	 * @return
+	 */
+	PublicKey getPublicKey(String alias);
+
+	// TODO is it OK to expose password like this? or should I specify password in

```

Comment:
this is "only" the keystore password, has nothing to do with user passwords right? what do you mean by "should I specify password in config for each class that needs to access the 'underlayingkeystore'"? lets discuss this solution a bit cause i dont really get whats happening

Reply:
This is a leftover comment from the first implementer who also wanted to discuss this, it seems. I think we're good to delete the comments.